test/operations_test/operations_test.go

```diff
@@ -257,12 +257,78 @@ var _ = test.DescribeTestsFor(test.TestCase{
 					ctxBuilder = NewTestContextBuilderWithSecurity().WithEnvPostExtensions(postHook)
 				})
 
+				When("A resource-less operation is created", func() {
+					BeforeEach(func() {
+						ctx = ctxBuilder.Build()
+						resourcelessOperation := types.Operation{
+							Base:       types.Base{ID: "test-resource-less-operation"},
+							ResourceID: "non-existent-resource",
+							Type:       types.CREATE,
+							State:      "succeeded",
+						}
+						ctx.SMRepository.Create(context.Background(), &resourcelessOperation)
+
+					})
+					It("Should be deleted once cleanup interval has passed", func() {
+						ensureOperationExist(ctx.SMRepository, "test-resource-less-operation")
+						Eventually(func() types.Object {
+							byID := query.ByField(query.EqualsOperator, "id", "test-resource-less-operation")
+							resourcelessOperation, _ := ctx.SMRepository.Get(context.Background(), types.OperationType, byID)
+							return resourcelessOperation
+						}, cleanupInterval*4).Should(BeNil())
+
+					})
+				})
+
+				When("Internal Operation is the 'last' for a given resource", func() {
+					BeforeEach(func() {
+						ctx = ctxBuilder.Build()
+						testResource1 := &types.Platform{
+							Base: types.Base{ID: "test-resource1"},
+						}
+						testResource2 := &types.Platform{
+							Base: types.Base{ID: "test-resource2"},
+						}
+						lastInternalSuccessfulOperation := &types.Operation{
+							Base:       types.Base{ID: "test-last-op-internal-successful"},
+							ResourceID: testResource1.ID,
+							Type:       types.CREATE,
+							State:      "succeeded",
+						}
+						lastInternalFailedOperation := &types.Operation{
+							Base:       types.Base{ID: "test-last-op-internal-failed"},
+							ResourceID: testResource2.ID,
+							Type:       types.CREATE,
+							State:      "succeeded",
+						}
+						ctx.SMRepository.Create(context.Background(), testResource1)
+						ctx.SMRepository.Create(context.Background(), testResource2)
+						ctx.SMRepository.Create(context.Background(), lastInternalSuccessfulOperation)
+						ctx.SMRepository.Create(context.Background(), lastInternalFailedOperation)
+					})
+					It("Should not be deleted", func() {
+						ensureOperationExist(ctx.SMRepository, "test-last-op-internal-successful")
+						ensureOperationExist(ctx.SMRepository, "test-last-op-internal-failed")
+
+						Eventually(func() bool {
```

Comment:
Eventually is good when the expected state is not the current state. Calling it like this doesn't really test anything - you ensure that this is the state before the "Eventually", so it is immediately fulfilled. If the last ops will be deleted the test will miss it.